### ISSUE TITLE
Use symlink instead of nesting scripts for BUILD_PREFIX/bin/python

### ISSUE THREAD

**Issue by hmaarrfk** [conda-forge/cross-python-feedstock#64]
In https://github.com/conda-forge/tensorflow-feedstock/pull/286#issuecomment-1360865890 @pkgw found that nesting scripts isn't really ideal for OSX:

> My testing indicates that on Linux, you can "nest" shebang scripts in this way and the sensible thing will happen, but on OSX it appears that is not the case. See [this StackOverflow Q&A](https://stackoverflow.com/questions/25894895/why-does-osx-require-shebang-interpreters-to-be-binaries).

https://github.com/conda-forge/cross-python-feedstock/blob/c356bb3a8fc9d98bf6fa8ae6c4fe3ca8e33ec9d6/recipe/activate-cross-python.sh#L53-L54

Could we make this a symlink instead?

**Comment by pkgw** [comment]
It sounds like this issue especially affects Cython, as per https://github.com/conda-forge/tensorflow-feedstock/pull/286#issuecomment-1350192505.

I presume there's a good reason for it, but is there anything written down explaining why the build-prefix `python` is delegating to the host-prefix `python`? That's hard for me to understand.

**Comment by pkgw** [comment]
Digging a bit more, I think it won't work to *just* change the wrapper script to a symlink, because if I'm reading things correctly, `$PREFIX/bin/python` is *also* a shell script, set up here:

https://github.com/conda-forge/cross-python-feedstock/blob/c356bb3a8fc9d98bf6fa8ae6c4fe3ca8e33ec9d6/recipe/activate-cross-python.sh#L38-L40

**Comment by pkgw** [comment]
For reference: this behavior was introduced in #7.

**Comment by hmaarrfk** [comment]
are you able to use debug mode for this? I've found that i had to reduce recipes to a "single output" before debugging effectively. Then i can run with 
```
conda debug -m .ci_support/osx_arm.yaml recipe
```

I also removed the parts of the code that force a bazel clean.

**Comment by pkgw** [comment]
So I've been thinking about how this issue could be addressed *without* a thinking-outside-of-the-box solution — if we take it as a given that we want the activator to continue providing executables in `$BUILD_PREFIX/bin/python` and $PREFIX/bin/python` that have the behavior of launching `$BUILD_PREFIX/venv/cross/bin/python`, but are binaries so that we don't run into this macOS nested-shebang issue.

I *think* it might work if this feedstock built a tiny C/Rust program that basically just did:

```
exec("$PREFIX/bin/python", "$PREFIX/venv/cross/bin/python", argv...)
```

installed it as something like `$PREFIX/bin/_cross_python_launcher`, and then the activation script copied that into `$BUILD_PREFIX/bin/python` and `$PREFIX/bin/python`. Since cross-python is getting installed as a `build:` dep, the prefix rewriting embeds the `$BUILD_PREFIX` for the binary stuff, and then if you copy that into `$PREFIX` the absolute paths will stay the same and you'll get the desired behavior.

That isn't really *that* complex but it still feels pretty gross and I'd be happy to think of an out-of-the-box solution that didn't require it. But if macOS can't nest shebang scripts, and the cross-python framework relies on a wrapper for `python` that sets up magic variables, it is hard for me to see a way to avoid distributing some kind of custom binary shim.

**Comment by isuruf** [comment]
> I think it might work if this feedstock built a tiny C/Rust program that basically just did:

This is what I had in mind. I started a branch doing this a year or so ago, but didn't really finish it and can't find the code.

**Comment by pkgw** [comment]
@isuruf If you think that idea is the way to go, I can look into implementing it!

**Comment by hmaarrfk** [comment]
An other manifestation of this bug that I found in one of my recipes:

@pkgw, I feel like @erykoff fixed up the zfpy recipe a while back with a symlink 
https://github.com/conda-forge/zfpy-feedstock/pull/22

**Comment by pkgw** [comment]
I've prototyped a solution with a shim binary program. Took me a bit longer than expected to get it going, but I'm currently testing it with the tensorflow build. Pull request if/when I can validate that it seems to work.

**Comment by pkgw** [comment]
Posted my solution in #65. I haven't yet successfully built all of Tensorflow but I feel confident that I've solved the issues that led to the opening of this issue.

**Comment by isuruf** [comment]
Fixed by #66
